aviatrix-controller-initialize/main.tf

```diff
@@ -18,9 +22,37 @@ resource "aws_iam_role" "iam_for_lambda" {
 EOF
 }
 
+resource "aws_iam_policy" "lambda-policy" {
+  name        = "${local.name_prefix}aviatrix-lambda-policy"
+  path        = "/"
+  description = "Policy for creating aviatrix-lambda-policy"
+  policy      = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeInstances",
+        "ec2:CreateNetworkInterface",
+        "ec2:AttachNetworkInterface",
+        "ec2:DescribeNetworkInterfaces",
+        "ec2:DeleteNetworkInterface",
+        "ec2:DetachNetworkInterface",
+        "ec2:ModifyNetworkInterfaceAttribute",
+        "ec2:ResetNetworkInterfaceAttribute",
+        "autoscaling:CompleteLifecycleAction"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
 resource "aws_iam_role_policy_attachment" "attach-policy" {
   role       = aws_iam_role.iam_for_lambda.name
-  policy_arn = "arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy"
+  policy_arn = aws_iam_policy.lambda-policy.arn
```

Comment:
Does the destroy work with the regular CloudWatchAgentServerPolicy policy?

Reply:
Regular CloudWatchAgentServerPolicy policy does not have permissions to call CreateNetworkInterface on EC2, so changes are needed.

Reply:
ok